### PR TITLE
DEVPROD-667 remove private projects

### PR DESCRIPTION
**Pull request by ablack12**
DEVPROD-667
### Description
Remove logic to allow for the existence of public projects.

I didn't remove the private variable or the auto-defaulting to private, to maintain backwards compatibility in case we have to revert this change, and then some projects were set to public. I'll deploy that change a couple of days after this one. 

### Testing
Will verify in staging that I'm able to access things as a logged in user. 

### Documentation
Removed references.

### CODE REVIEW

**Comment by ablack12** [comment]
evergreen refresh

File: rest/route/middleware.go [hunk 1]
```diff
@@ -594,62 +593,35 @@ func updateHostAccessTime(ctx context.Context, h *host.Host) {
 }
 
 func RequiresProjectPermission(permission string, level evergreen.PermissionLevel) gimlet.Middleware {
```
Comment (malikchaya2) [inline rest/route/middleware.go:595]: We have a few routes in [rest/route/service.go](https://github.com/malikchaya2/evergreen/blob/main/rest/route/service.go#L112)
 that have just viewTasks or which maps to this. With this change, I think we should change them from `( viewTasks)` to `(requireUser, viewTasks)`.

Comment (ablack12) [reply rest/route/middleware.go:595]: viewTasks already does verify user in urlVarsToProjectScopes but can add it more explicitly.

**Review by malikchaya2 — APPROVED** [review]
LGTM once `Will verify in staging that I'm able to access things as a logged in user` is done

**Pull request merged**